product/products.go

```diff
@@ -37,6 +37,18 @@ type Product struct {
 	Selects  []string
 }
 
+// Statuses reads from each seeker on the product and returns a map containing sums of each Status
+func (p *Product) Statuses() (map[seeker.Status]int, error) {
+	statuses := make(map[seeker.Status]int)
+	for _, s := range p.Seekers {
+		if s.Status == "" {
+			return nil, fmt.Errorf("unable to build Statuses map, seeker not run: seeker=%s", s.Identifier)
```

Comment:
Yeah I thought nil-ing empty collections was pretty weird the first time I read it. My bunch is that the  [wiki](https://github.com/golang/go/wiki/CodeReviewComments#declaring-empty-slices) chooses nil because of the developer ergonomics. Most of the time errors are handled in guards so the result is never used - though there's several places in hcdiag where we want to save the result anyway even if a command errors.